src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { peopleFromServer } from '../data/people';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+type Props = {
+  onSelected?: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [list, setList] = useState<boolean>(false);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const filteredList = useMemo(() => {
+    return peopleFromServer.filter(people =>
+      people.name
+        .toLocaleLowerCase()
+        .trim()
+        .includes(appliedQuery.toLocaleLowerCase().trim()),
+    );
+  }, [appliedQuery]);
+
+  const selectedPerson = (person: Person) => {
+    setQuery(person?.name);
```

Comment:
Optional chaining (`?.`) is unnecessary here because `person` is always expected to be a valid `Person` object when `selectedPerson` is called. You can safely use `person.name`.